packages/gpa/test_utils.go

```diff
@@ -11,9 +11,9 @@ import (
 func MakeTestNodeIDFromIndex(index int) NodeID {
 	nodeID := NodeID{}
 
-	indexBytes := make([]byte, 8)
-	binary.LittleEndian.PutUint64(indexBytes, uint64(index))
-	copy(nodeID[:8], indexBytes)
+	indexBytes := make([]byte, 4)
+	binary.BigEndian.PutUint32(indexBytes, uint32(index))
+	copy(nodeID[:4], indexBytes)
```

Comment:
Can we also copy the bytes at the end of "nodeID" ?
```suggestion
	copy(nodeID[NodeIDLength-4:NodeIDLength], indexBytes)
```
And also change the ShortString method to use these bytes?
Otherwise we have  `0x000000010000000000000000000000000000000000000000000000000000000` as the "full node ID" instead of `0x000000000000000000000000000000000000000000000000000000000000001`

Reply:
Well... `ShortString` always takes first 4 bytes. (see e.g. `ChainID`). To be consistent, `ShortString` for `NodeID` also takes first 4 bytes. And this `NodeID` generation is just for tests, so it adapts to the real code (`ShortString`). Besides, the full ID is usually not logged.